File: EXERCICIOS/Lista 1/Questao2.java
import java.util.List;
public class Questao2 {

    public static void main(String[] args) {
        int dado01 = roladado();
        int dado02 = roladado();
        int pontos = dado01 + dado02;

        System.out.printf("Dado01: %d, Dado02: %d\n", dado01, dado02);
        System.out.printf("pontos: %d\n", pontos);

        if (List.of(7, 11).contains(pontos)) {
            System.out.println("VOCÊ GANHOU!!!");
            return;
        }

        if (List.of(2, 3, 12).contains(pontos)) {
            System.out.println("PERDEU!!!");
            return;
        }

        System.out.println("PRÓXIMA FASE!");

        int contador = 0;
        while (true) {
            contador++;

            dado01 = roladado();
            dado02 = roladado();
          int soma = dado01 + dado02;

            System.out.printf("Dice 1: %d, Dice 2: %d\n", dado01, dado02);
            System.out.printf("Play %d: %d\n", contador, soma);

            if (soma == 7) {
                System.out.println("VOCÊ PERDEU!!!");
                return;
            }

            if (soma == pontos) {
                System.out.println("VOCÊ GANHOU!!!");
                return;
            }
        }
    }

    private static int roladado() {
        return (int) (Math.random() * 6) + 1;
    }
}

File: EXERCICIOS/Lista 2/Questao2.py
import random

def jogar_dados():
    return random.randint(1, 6)

def main():
    dado01 = jogar_dados()
    dado02 = jogar_dados()
    pontos = dado01 + dado02

    print(f"Dado 01: {dado01}, Dado 02: {dado02}")
    print(f"Pontos: {pontos}")

    if pontos in (7, 11):
        print("VOCÊ GANHOU!!!")
        return

    if pontos in (2, 3, 12):
        print("PERDEU!!!")
        return

    print("PRÓXIMA FASE!")

    contador = 0
    while True:
        contador += 1

        dado01 = jogar_dados()
        dado02 = jogar_dados()
        soma = dado01 + dado02

        print(f"Dado 1: {dado01}, Dado 2: {dado02}")
        print(f"Jogada {contador}: {soma}")

        if soma == 7:
            print("VOCÊ PERDEU!!!")
            return

        if soma == pontos:
            print("VOCÊ GANHOU!!!")
            return

if __name__ == "__main__":
    main()
